homeassistant/components/wmspro/scene.py

```diff
@@ -0,0 +1,60 @@
+"""Support for scenes provided by WMS WebControl pro."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from wmspro.scene import Scene as WMS_Scene
+
+from homeassistant.components.scene import Scene
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WebControlProConfigEntry
+from .const import ATTRIBUTION, DOMAIN, MANUFACTURER
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: WebControlProConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the WMS based scenes from a config entry."""
+    hub = config_entry.runtime_data
+
+    entities: list[WebControlProScene] = [
+        WebControlProScene(config_entry.entry_id, scene)
+        for scene in hub.scenes.values()
+    ]
+
+    async_add_entities(entities)
+
+
+class WebControlProScene(Scene):
+    """Representation of a WMS based scene."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, config_entry_id: str, scene: WMS_Scene) -> None:
+        """Initialize the entity with the configured scene."""
+        super().__init__()
+        scene_id_str = str(scene.id)
+        self._scene = scene
+        self._attr_unique_id = scene_id_str
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, scene_id_str)},
+            manufacturer=MANUFACTURER,
+            model="Scene",
+            name=scene.name,
+            serial_number=scene_id_str,
+            suggested_area=scene.room.name,
+            via_device=(DOMAIN, config_entry_id),
+            configuration_url=f"http://{scene.host}/control",
+        )
```

Comment:
Why is every scene its own device?

Reply:
It doesn't need to be. Scenes are not a device in the real world, so we could skip this part.

Reply:
But I am interested in how they work since they are linked to a room

Reply:
On the automatically created dashboard it puts them into the corresponding room:
<img width="595" alt="image" src="https://github.com/user-attachments/assets/f55d37d7-f706-472b-8a6c-3e894e7fb6f5">

Reply:
It works the same as with other entities which have device information:
<img width="1088" alt="image" src="https://github.com/user-attachments/assets/1c797091-d97e-4644-bac1-6e4d0d2c342c">

Reply:
yeye, I get that, but how does Scenes work in the world of WMS

Reply:
It is basically just a one-off trigger of saved target states. You configure them like this:
<img width="1118" alt="image" src="https://github.com/user-attachments/assets/296fa31b-86dc-4d2e-ae77-7c5862ef23f2">
You can add any WMS device/destination there and set a desired target state.

Reply:
Scenes in WMS are basically the same thing as in [HA](https://developers.home-assistant.io/docs/core/entity/scene/): "A scene entity is an entity that can reproduce a wanted state for a group of entities. A scene entity can activate the scene towards a group of devices but remains stateless ..."

Reply:
Check. Yea I remember that Hue also has scenes, but they are coupled to the group they affect

Reply:
Okay, so should I remove or keep the device info?

Reply:
I just checked: The following integrations also provide DeviceInfo for Scene implementations:
- fibaro
- hue
- litejet
- lutron_caseta
- tuya

Reply:
I did some more research and found out the following:

**Fibaro** links the scenes to the **hub** device:
https://github.com/home-assistant/core/blob/8950e817e0a69ba2eb61987cd144cee4f31537b7/homeassistant/components/fibaro/scene.py#L51-L54

Hue actually links the scenes to the **room/zone** device:
https://github.com/home-assistant/core/blob/8950e817e0a69ba2eb61987cd144cee4f31537b7/homeassistant/components/hue/scene.py#L103-L107

**Litejet** links the scenes to the **hub** (mcp?) device:
https://github.com/home-assistant/core/blob/8950e817e0a69ba2eb61987cd144cee4f31537b7/homeassistant/components/litejet/scene.py#L51-L56

**Lutron Caseta** links the scenes to the **hub** (bridge) device:
https://github.com/home-assistant/core/blob/8950e817e0a69ba2eb61987cd144cee4f31537b7/homeassistant/components/lutron_caseta/scene.py#L41-L43

Only **Tuya** links the scene to its own (scene_id) device:
https://github.com/home-assistant/core/blob/8950e817e0a69ba2eb61987cd144cee4f31537b7/homeassistant/components/tuya/scene.py#L37
https://github.com/home-assistant/core/blob/8950e817e0a69ba2eb61987cd144cee4f31537b7/homeassistant/components/tuya/scene.py#L44-L50

So, we have **Hue** linking to rooms, 3 others linking to their hub and just **Tuya** creating per scene devices.

Based on these findings, any suggestion on how to move forward?